### PR TITLE
feat: prepare for release v0.7.0

### PR DESCRIPTION
**Pull request by foxpy**
- bump workspace_optimizer to v0.14.0
- bump Rust dependencies and organize them into workspace
- set release profile at workspace Cargo.toml
- bump Rust toolchain to v1.71.0
- prepare shell scripts for sdk v47: use broadcast mode sync

### CODE REVIEW

File: scripts/test_ibc_transfer.sh [hunk 1]
```diff
@@ -10,31 +10,43 @@ NEUTRON_DIR="${NEUTRON_DIR:-../../neutron}"
 HOME="$NEUTRON_DIR/data/test-1/"
 KEY="demowallet1"
 ADMIN="neutron1m9l358xunhhwds0568za49mzhvuxx9ux8xafx2"
-BIN="neutrond"
```
Comment (pr0n00gler) [inline scripts/test_ibc_transfer.sh:13]: What's the reason for removing `BIN` and `GAIA_BIN` constants?

Comment (foxpy) [reply scripts/test_ibc_transfer.sh:13]: what's the original reason for them to be defined as a constant?

Comment (pr0n00gler) [reply scripts/test_ibc_transfer.sh:13]: Because i don't need to change binary name everywhere if it's changed for some reason

Comment (foxpy) [reply scripts/test_ibc_transfer.sh:13]: And how often do we see them changing? Also, you will have to check and replace these constants in every single script anyway, hence I didn't see much value in them

File: packages/neutron-sdk/src/interchain_txs/helpers.rs [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
-    let msg_data: Result<TxMsgData, DecodeError> = TxMsgData::decode(data.as_slice());
+    let msg_data: Result<TxMsgData, _> = TxMsgData::decode(data.as_slice());
     match msg_data {
         Err(e) => Err(StdError::generic_err(format!(
             "Can't decode response: {}",
             e
         ))),
-        Ok(msg) => Ok(msg.data),
+        Ok(msg) => Ok(msg
+            .msg_responses
```
Comment (foxpy) [inline packages/neutron-sdk/src/interchain_txs/helpers.rs:16]: this field should only be used starting from sdk46, currently neutron uses sdk45, hence we have to use deprecated field msg.data for a while, until sdk47 upgrade is live

**Comment by foxpy** [comment]
Successful test run: https://github.com/neutron-org/neutron-tests/actions/runs/6454109468

**Comment by foxpy** [comment]
Successful test run: https://github.com/neutron-org/neutron-tests/actions/runs/6532055037

**Review by pr0n00gler — APPROVED** [review]

**Review by sotnikov-s — APPROVED** [review]

**Pull request merged**